R/mail.R

```diff
@@ -6,110 +6,249 @@
+#' Refresh mbox files downloaded via pipermail
+#' Uses the adopted file name convention by \code{\link{download_pipermail}} to identify
+#' the latest downloaded mbox. It deletes this file, then redownloads it along with all future months
+#' up to the current real-life month.
+#' If the directory is empty, then it downloads all pipermail files (as mbox files) via \code{\link{download_pipermail}}
+#' @param mailing_list The name of the mailing list being downloaded
+#' @param start_year_month The year and month of the first file to be downloaded
+#' @param save_folder_path The folder path in which all the downloaded pipermail files will be stored
+#' @return Returns `downloaded_files`, a vector of the downloaded files in the current working directory
 #' @export
-convert_pipermail_to_mbox <- function(filelist) {
+refresh_pipermail <- function(mailing_list, start_year_month, save_folder_path) {
 
-  #at to @ replace function
-  pipermail_atreplacer <- function(string) {
+  # Create directory if it does not exist
+  if (!dir.exists(save_folder_path)) {
+    dir.create(save_folder_path, recursive = TRUE)
+  }
 
-    rstring <- sub(" at ", "@", string)
+  # Check if the folder is empty
+  files_in_folder <- list.files(save_folder_path, pattern = "kaiaulu_\\d{6}\\.mbox$")
+  if (length(files_in_folder) == 0) {
+    # If empty, download from start_year_month to the current month
+    end_year_month <- format(Sys.Date(), "%Y%m")
+    cat("Folder is empty. Downloading from", start_year_month, "to", end_year_month, "\n")
+    download_pipermail(mailing_list, start_year_month, end_year_month, save_folder_path)
+    return(NULL)
+  }
+  # If folder is not empty, find the most recent month
+  year_months <- gsub("kaiaulu_(\\d{6})\\.mbox$", "\\1", files_in_folder)
+  recent_month <- max(year_months)
+
+  # Delete the most recent file
+  recent_file <- file.path(save_folder_path, paste0("kaiaulu_", recent_month, ".mbox"))
+  if (file.exists(recent_file)) {
+    file.remove(recent_file)
+    cat("Deleted the most recent file:", recent_file, "\n")
+  }
 
-    return(rstring)
+  # Redownload from the most recent month to the current real-life month
+  end_year_month <- format(Sys.Date(), "%Y%m")
+  cat("Redownloading from", recent_month, "to", end_year_month, "\n")
+  download_pipermail(mailing_list, recent_month, end_year_month, save_folder_path)
+}
 
-  }
 
-  output <- "output.mbox"
+#' Process .gz files in a folder, unzip and convert them to .mbox
+#' Checks a folder for any .gz files, unzips them, and renames them
+#' to .mbox format. The original .gz files are deleted after unzipping. If a .mbox
+#' file with the same name already exists, it will be overwritten.
+#'
+#' @param folder_path The path to the folder containing both .gz and .mbox files.
+#' @return A list of the .mbox files that were created or updated.
+#' @export
+process_gz_to_mbox_in_folder <- function(folder_path) {
```

Comment:
this function is not currently called on the refresh_ function. It should be called on every refresh call to ensure the ending state of the folder on refresh is always preserved with .mbox. I would not add that logic to download_pipermail(), in case the user wants to have the original files, they can use download_pipermail() directly.